pkg/lang/csharp/declarations.go

```diff
@@ -0,0 +1,662 @@
+package csharp
+
+import (
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/filter/predicate"
+	"github.com/klothoplatform/klotho/pkg/logging"
+	"github.com/klothoplatform/klotho/pkg/query"
+	sitter "github.com/smacker/go-tree-sitter"
+	"go.uber.org/zap"
+	"strings"
+)
+
+/* TODO: implement support for:
+   - properties
+   - constructors
+   - events
+   - enums
+   - indexers
+   - local functions
+   - local variables
+   - delegates
+*/
+
+type Declaration struct {
```

Comment:
The line for where to break up files in go is pretty blurry, but we can definitely break the file down into smaller chunks. The booleans for all the different modifiers are definitely a bit noisy and can be minimized (aside from visibility).

Having the ability to parse all the declarations in a file is crucial for understanding dependencies between files since C# imports are always either `namespaces` or specific types -- never files.

Additionally, parsing declarations into a higher level format makes them much easier to work with in plugins as opposed to dealing with one-off queries each time we need to find and work with a particular declaration.

Reply:
> one-off queries each time we need to find and work with a particular declaration

Hm, generally I prefer to work with one-off queries since they're easier to iterate and test with in addition to being a fair amount more succinct than doing the same in Go code. I won't block since this is just my opinion, but maybe we can discuss as a team so we can have a consistent stance on it.